osutil/epoll/epoll.go

```diff
@@ -91,19 +125,21 @@ func (e *Epoll) decrementRegisteredFdCount() {
 	atomic.AddInt32(&e.registeredFdCount, -1)
 }
 
-func (e *Epoll) zeroRegisteredFdCount() {
-	atomic.StoreInt32(&e.registeredFdCount, 0)
-}
-
 // Register registers a file descriptor and allows observing speicifc I/O readiness events.
 //
 // Please refer to epoll_ctl(2) and EPOLL_CTL_ADD for details.
 func (e *Epoll) Register(fd int, mask Readiness) error {
+	if e.IsClosed() {
+		return ErrEpollClosed
+	}
 	e.incrementRegisteredFdCount()
 	err := unix.EpollCtl(e.fd, unix.EPOLL_CTL_ADD, fd, &unix.EpollEvent{
 		Events: uint32(mask),
 		Fd:     int32(fd),
 	})
+	if e.IsClosed() {
```

Comment:
I think conceptually all of these should be: if err != nil && e.IsClosed()

Reply:
That makes sense, thank you